### DISCUSSION TITLE
[V5] - Select AS ORM queries

### DISCUSSION THREAD

**Discussion by sn0rcha** [adonisjs/core#2611] [Help] [answered]
Hey Guys,

Has anyone had any luck doing select AS queries with an ORM model in V5. The following doesn't work:

```
SomeModel.query().select({label:'name'})
```
returns
```
[{}]
```

This works however

```
SomeModel.query().select({name:'name'})
```
returns

```
[{name:'Some Name'}]
```
Any ideas - "name" is the database field defined in the model.

**Comment by thetutlage** [comment, marked as answer]
May I know the reason behind aliasing the field name when using models? Because this defeats the entire point of using models

**Reply by sn0rcha** [reply]
sometimes I need to select a field as a different key ("label" in this case) for front end frameworks that build select menus automatically based on certain keys existing in the array (Quasar for example). I was reading this from the docs - https://docs.adonisjs.com/reference/database/query-builder#select

**Reply by thetutlage** [reply]
Since you are creating an API server, you should be using the [serialization layer](https://docs.adonisjs.com/guides/models/serializing-models) to rename the model properties.

If it is just for one query, then that is not the right approach. If you read about API resources in general, then you will find that they should have a consistent shape and representation and not something that changes between HTTP requests.

Now to answer the main question. "Where is the select alias?". It is under the `$extras` object, since that property name is not defined on the model as a column, hence it has been moved to the `$extras` object. More in docs https://docs.adonisjs.com/guides/models/serializing-models#serializing-extras-object

So you can access it as 

```ts
const results = await SomeModel.query().select({label:'name'})
results[0].$extras.label
```

**Reply by sn0rcha** [reply]
Thanks for the info much appreciated!
